### PR TITLE
HL-385: validate start_date when submitting a draft application

### PR DESCRIPTION
**Pull request by haavikko**
## Description :sparkles:

start_date on a previous year prevented saving a draft application.

## Testing :alembic:
backend/benefit/applications/tests/test_applications_api.py: test_application_date_range_on_submit

### CODE REVIEW

File: backend/benefit/applications/tests/test_applications_api.py [hunk 1]
```diff
@@ -716,6 +721,40 @@ def test_application_date_range(
         assert response.data["end_date"] == end_date
 
 
+@pytest.mark.parametrize(
```
Comment (karisal-anders) [inline backend/benefit/applications/tests/test_applications_api.py:724]: Maybe freezing the time for this test would be good, e.g.
`@freeze_time("2021-01-01")`?

**Review by karisal-anders — COMMENTED** [review]
Looks ok to me except that `test_application_date_range_on_submit` probably needs time freezing and please could you doublecheck whether `test_application_date_range` needs it? Thank you!

File: backend/benefit/applications/tests/test_applications_api.py (hunk 1 above)
Comment (haavikko) [reply backend/benefit/applications/tests/test_applications_api.py:724]: freeze_time is applied to all these tests automatically (from common.tests.conftest)

**Review by karisal-anders — APPROVED** [review]

File: backend/benefit/applications/tests/test_applications_api.py (hunk 1 above)
Comment (karisal-anders) [reply backend/benefit/applications/tests/test_applications_api.py:724]: Ah, ok so it was `from common.tests.conftest import *` in `backend/benefit/applications/tests/test_applications_api.py` which imported the autouse fixture `setup_test_environment` from `common.tests.conftest` which made the freeze_time be applied to these tests. Alright!

**Pull request merged**